articles/flutter-vs-reactnative/index.md

```diff
@@ -0,0 +1,113 @@
+---
+layout: engineering-education
+status: publish
+published: true
+url: /engineering-education/flutter-vs-reactnative/
+title: Flutter vs React Native - Which is a better cross platform framework
+description:
+author:
+date: 2020-10-18T00:00:00-14:00
+topics: [Languages]
+excerpt_separator: <!--more-->
+images:
+
+  - url: /engineering-education/flutter-vs-reactnative/hero.jpg
+    alt:
+---
+Companies today are not just making products, but ecosystems. A gamut of services which can prove to be a steady revenue stream. It is only natural for them to try to get apps to run across devices within those ecosystems. How about being able to do that with a single codebase? Wishful thinking? That may have been true 20 years ago, but not today.
+<!--more-->
+#### Take 1: One bytecode to rule ‘em all
+The dream of a cross-platform was first braved by Java. It devised the paradigm of [Write Once Run Anywhere (WORA)](https://en.wikipedia.org/wiki/Write_once,_run_anywhere). It was successful to a major extent, according to the Java installer at least.
+
+![](/engineering-education/flutter-vs-reactnative/installer.jpeg)
+
+Things went south when companies started making their own Java implementations for exclusivity. Oracle turned licenses and usage rights upside down. Hundreds of products like J2ME, J2EE, Open JDK, Jakarta etc. popped up. Java became too fragmented to be universal.
+
+#### Take 2: A glorified webpage
+Mobile devices became more powerful, more compact, and more connected. They were getting more attention than our humble PC or laptops did. So companies were looking to develop for mobile while supporting PC software applications. Due to mobile being the future, but the PC market still having one of the largest user base. The thought process was to present a webpage wrapped in an app. Frameworks like ionic, PhoneGap, and Cordova tried to do it but the performance was not on par.
```

Comment:
links to the frameworks (or links to more info) presented in the sentence would help the audience